incubator/hnc/pkg/controllers/config_controller_test.go

```diff
@@ -0,0 +1,37 @@
+package controllers_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("HNCConfiguration", func() {
+	ctx := context.Background()
+
+	var (
+		fooName string
+		barName string
+	)
+
+	BeforeEach(func() {
+		fooName = createNS(ctx, "foo")
+		barName = createNS(ctx, "bar")
+	})
+
+	It("should create a new ObjectReconciler when adding a new type to the HNCConfiguration", func() {
```

Comment:
You're not testing that an ObjectReconciler exists - you're testing that the objects get propagated. So maybe:

`It("should propagate objects whose types have been added to HNCConfiguration")`

Reply:
Thanks for the suggestion. Changed the description. 

I also reset the singleton to default in AfterEach in object_controller_test.go and config_controller_test.go as we discussed. Feel free to take a look!

Thanks so much for the discussion!